### PR TITLE
feat: add transaction_by_hash_with_meta

### PR DESCRIPTION
**Pull request by mattsse**
adds a new provider function and type to fetch transaction related metadata required for `transactionByHash` and the receipt

### CODE REVIEW

File: crates/storage/provider/src/providers/mod.rs [hunk 1]
```diff
@@ -181,6 +181,50 @@ impl<DB: Database> TransactionsProvider for ShareableDatabase<DB> {
...
+                    if let Some(transaction) = tx.get::<tables::Transactions>(transaction_id)? {
+                        let mut transaction_cursor =
+                            tx.cursor_read::<tables::TransactionBlock>()?;
+                        if let Some(block_number) =
+                            transaction_cursor.seek(transaction_id).map(|b| b.map(|(_, bn)| bn))?
+                        {
+                            if let Some(block_hash) =
+                                tx.get::<tables::CanonicalHeaders>(block_number)?
+                            {
+                                if let Some(block_body) =
```
Comment (onbjerg) [inline crates/storage/provider/src/providers/mod.rs:200]: I wonder if this would be nicer using `Option::and_then` and `Option::map`?

**Review by onbjerg — APPROVED** [review]
Small comment on potentially making it a bit nicer, otherwise LGTM

File: crates/storage/provider/src/providers/mod.rs [hunk 2]
```diff
@@ -181,6 +181,50 @@ impl<DB: Database> TransactionsProvider for ShareableDatabase<DB> {
             .map_err(Into::into)
     }
 
+    fn transaction_by_hash_with_meta(
+        &self,
+        tx_hash: TxHash,
+    ) -> Result<Option<(TransactionSigned, TransactionMeta)>> {
+        self.db
+            .view(|tx| -> Result<_> {
+                if let Some(transaction_id) = tx.get::<tables::TxHashNumber>(tx_hash)? {
```
Comment (onbjerg) [inline crates/storage/provider/src/providers/mod.rs:190]: Use `transaction_block` defined in this trait

**Review by onbjerg — CHANGES_REQUESTED** [review]
Nvm one small nit here on using some of the built-in functions of the trait

File: crates/storage/provider/src/providers/mod.rs (hunk 1 above)
Comment (mattsse) [reply crates/storage/provider/src/providers/mod.rs:200]: I tried this first, total mess

**Pull request merged**